front50-web/src/main/java/com/netflix/spinnaker/front50/controllers/PipelineController.java

```diff
@@ -237,33 +258,106 @@ public Pipeline getByApplicationAndName(
   }
 
   @PreAuthorize(
-      "@fiatPermissionEvaluator.storeWholePermission() and hasPermission(#pipeline.application, 'APPLICATION', 'WRITE') and @authorizationSupport.hasRunAsUserPermission(#pipeline)")
+      "@fiatPermissionEvaluator.storeWholePermission() "
+          + "and hasPermission(#pipeline.application, 'APPLICATION', 'WRITE') "
+          + "and @authorizationSupport.hasRunAsUserPermission(#pipeline)")
   @RequestMapping(value = "", method = RequestMethod.POST)
   public synchronized Pipeline save(
       @RequestBody Pipeline pipeline,
       @RequestParam(value = "staleCheck", required = false, defaultValue = "false")
           Boolean staleCheck) {
 
-    validatePipeline(pipeline, staleCheck);
+    long saveStartTime = System.currentTimeMillis();
+    log.info(
+        "Received request to save pipeline {} in application {}",
+        pipeline.getName(),
+        pipeline.getApplication());
 
+    log.debug("Running validation before saving pipeline {}", pipeline.getName());
+    long validationStartTime = System.currentTimeMillis();
+    validatePipeline(pipeline, staleCheck);
+    checkForDuplicatePipeline(
+        pipeline.getApplication(), pipeline.getName().trim(), pipeline.getId());
     pipeline.setName(pipeline.getName().trim());
```

Comment:
This is being done in validatePipeline so don't think this is needed here anymore.

Reply:
agreed. Will change.